src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -104,60 +104,112 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 }
 
 static void
-boot_newcomp_create(spdid_t spdid, struct cos_compinfo *comp_info)
+boot_newcomp_definfo_init(spdid_t spdid, compcap_t cc, int is_sched)
 {
-	compcap_t   cc;
-	captblcap_t ct = new_comp_cap_info[spdid].compinfo->captbl_cap;
-	pgtblcap_t  pt = new_comp_cap_info[spdid].compinfo->pgtbl_cap;
-	sinvcap_t   sinv;
-	thdcap_t    main_thd;
-	int         i = 0;
-
-	cc = cos_comp_alloc(&boot_info, ct, pt, (vaddr_t)new_comp_cap_info[spdid].upcall_entry);
-	assert(cc);
-	new_comp_cap_info[spdid].compinfo->comp_cap = cc;
 
-	/* Create sinv capability from Userspace to Booter components */
-	sinv = cos_sinv_alloc(&boot_info, boot_info.comp_cap, (vaddr_t)__inv_test_entry);
-	assert(sinv > 0);
+	struct cos_defcompinfo *defci     = cos_defcompinfo_curr_get();
+	struct cos_aep_info *   sched_aep = cos_sched_aep_get(defci);
+	struct cos_aep_info *   child_aep = cos_sched_aep_get(new_comp_cap_info[spdid].defcompinfo);
+	struct cos_compinfo *   child_ci  = cos_compinfo_get(new_comp_cap_info[spdid].defcompinfo);
+
+	child_ci->comp_cap = cc;
+	child_aep->thd = cos_initthd_alloc(boot_info, child_ci->comp_cap);
+
+	if (is_sched) {
+		assert(child_aep->thd);
+		child_aep->tc = cos_tcap_alloc(boot_info);
+		assert(child_aep->tc);
+
+		child_aep->rcv = cos_arcv_alloc(boot_info, child_aep->thd, child_aep->tc, boot_info->comp_cap, sched_aep->rcv);
+		assert(child_aep->rcv);
+	}
+
+	else {
+		child_aep->tc  = sched_aep->tc;
+		child_aep->rcv = sched_aep->rcv;
+	}
+
+	child_aep->fn   = NULL;
+	child_aep->data = NULL;
+}
```

Comment:
This is what is done in I think `sl_thd_comp_init or _alloc`. If it cannot be used directly for some reason, would be better to have a comment about that.